include/merlin/types.cuh

```diff
@@ -44,17 +44,34 @@ using byte = uint8_t;
 // Digest.
 using D = byte;
 
-constexpr uint64_t EMPTY_KEY = UINT64_C(0xFFFFFFFFFFFFFFFF);
-constexpr uint64_t RECLAIM_KEY = UINT64_C(0xFFFFFFFFFFFFFFFE);
-constexpr uint64_t VACANT_KEY_MASK = UINT64_C(0xFFFFFFFFFFFFFFFE);
-constexpr uint64_t LOCKED_KEY = UINT64_C(0xFFFFFFFFFFFFFFFD);
-constexpr uint64_t RESERVED_KEY_MASK = UINT64_C(0xFFFFFFFFFFFFFFFC);
 constexpr uint64_t MAX_SCORE = UINT64_C(0xFFFFFFFFFFFFFFFF);
 constexpr uint64_t EMPTY_SCORE = UINT64_C(0);
 constexpr uint64_t IGNORED_GLOBAL_EPOCH = UINT64_C(0xFFFFFFFFFFFFFFFF);
 
-#define IS_RESERVED_KEY(key) ((RESERVED_KEY_MASK & (key)) == RESERVED_KEY_MASK)
-#define IS_VACANT_KEY(key) ((VACANT_KEY_MASK & (key)) == VACANT_KEY_MASK)
+uint64_t EMPTY_KEY_CPU = UINT64_C(0xFFFFFFFFFFFFFFFF);
+__constant__ uint64_t EMPTY_KEY = UINT64_C(0xFFFFFFFFFFFFFFFF);
+__constant__ uint64_t RECLAIM_KEY = UINT64_C(0xFFFFFFFFFFFFFFFE);
+__constant__ uint64_t LOCKED_KEY = UINT64_C(0xFFFFFFFFFFFFFFFD);
+template <class K>
+__forceinline__ __device__ bool IS_RESERVED_KEY(K key) {
+  return key == EMPTY_KEY || key == RECLAIM_KEY || key == LOCKED_KEY;
```

Comment:
Considering the performance(IS_RESERVED_KEY will be called anytime on each threads), it would be better to use `&` operator, so we can use one reserved mask to indicate the reserved keys space like `RESERVED_KEY_MASK = UINT64_C(0xFFFFFFFFFFFFFFFC);`, by default, it should be 0x3, if we wanna reserve 0xFFFFFFFFFFFFFFxF(x = E, C, D), we can let the user give a mask valued 0x30.